src/components/collection/ResourceConfigForm.tsx

```diff
@@ -50,10 +50,40 @@ function ResourceConfigForm({ collectionName, readOnly = false }: Props) {
         },
     };
 
+    // Find field with x-collection-name annotation
+    const collectionNameFieldKey = useMemo(() => {
+        if (resourceSchema.properties) {
+            // Find the field with the collection name annotation
+            const collectionNameField =
+                Object.entries(resourceSchema.properties).find(
+                    ([_, value]) =>
+                        value && Annotations.defaultResourceConfigName in value
+                ) ?? [];
+
+            // Try to fetch the key
+            return collectionNameField[0];
+        }
+
+        return null;
+    }, [resourceSchema.properties]);
+
+    // Check if we need to add a default value
+    const preparedResourceSchema = useMemo(() => {
+        if (collectionNameFieldKey) {
+            // Add a default property set to the stripped collection name
+            const response = cloneDeep(resourceSchema);
+            response.properties[collectionNameFieldKey].default =
+                stripPathing(collectionName);
+            return response;
+        }
+
+        return resourceSchema.properties;
+    }, [collectionName, collectionNameFieldKey, resourceSchema]);
```

Comment:
Make a memo of the schema which will either be the normal schema or one with the injected default value.